### ISSUE TITLE
NullPointerException in "amazon.com" HEAD request

### ISSUE THREAD

**Issue by lopotun** [softwaremill/sttp#468]
When I issue **HEAD** request to http://www.amazon.com/ I get NullPointerException:
```
Exception in thread "main" java.lang.NullPointerException
	at sttp.client.internal.package$.transfer$1(package.scala:23)
	at sttp.client.internal.package$.transfer(package.scala:30)
	at sttp.client.internal.package$.toByteArray(package.scala:35)
	at sttp.client.HttpURLConnectionBackend.readResponseBody(HttpURLConnectionBackend.scala:260)
	at sttp.client.HttpURLConnectionBackend.readResponseBody(HttpURLConnectionBackend.scala:251)
	at sttp.client.HttpURLConnectionBackend.readResponse(HttpURLConnectionBackend.scala:240)
	at sttp.client.HttpURLConnectionBackend.$anonfun$send$1(HttpURLConnectionBackend.scala:61)
	at sttp.client.SttpClientException$.adjustSynchronousExceptions(SttpClientException.scala:48)
	at sttp.client.HttpURLConnectionBackend.adjustExceptions(HttpURLConnectionBackend.scala:289)
	at sttp.client.HttpURLConnectionBackend.send(HttpURLConnectionBackend.scala:29)
	at sttp.client.HttpURLConnectionBackend.send(HttpURLConnectionBackend.scala:23)
	at sttp.client.FollowRedirectsBackend.sendWithCounter(FollowRedirectsBackend.scala:29)
	at sttp.client.FollowRedirectsBackend.send(FollowRedirectsBackend.scala:17)
	at sttp.client.RequestT.send(RequestT.scala:280)
```
Here is the test code:
```
    def test() = {
      val url = "http://www.amazon.com/"
      import sttp.client._
      val request = basicRequest.head(uri"$url")
      val response = request.send()
      response.code
    }
```
If I issue **GET** request to http://www.amazon.com/ everything works fine.

**Comment by adamw** [comment]
Released in 0.5.11. Thanks for the report!

**Comment by adamw** [comment]
Released in 2.0.5
